subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/IdeaPlugin.java

```diff
@@ -458,6 +471,14 @@ private void setupScopes(Project project) {
         ideaModel.getModule().setScopes(scopes);
     }
 
+    private void configureIdeaModuleForTestSuites(final Project project) {
+        final TestingExtension testing = project.getExtensions().getByType(TestingExtension.class);
+        final IdeaModule ideaModule = ideaModelFor(project).getModule();
+        testing.getSuites().withType(JvmTestSuite.class).configureEach(suite -> {
+            ideaModule.getTestSourceDirs().addAll(suite.getSources().getAllSource().getSourceDirectories().getFiles());
```

Comment:
This doesn't play nice with the existing "convention mapping" found in this plugin, I think. We actually need to add to the `testSourceDirs` Set created in `configureIdeaModuleForJava`.  I think this may completely replace that and we could instead do the convention mapping here instead.